cli/login.go

```diff
@@ -136,18 +136,26 @@ func (r *RootCmd) login() *clibase.Cmd {
 		useTokenForSession bool
 	)
 	cmd := &clibase.Cmd{
-		Use:        "login <url>",
+		Use:        "login [<url>]",
 		Short:      "Authenticate with Coder deployment",
 		Middleware: clibase.RequireRangeArgs(0, 1),
 		Handler: func(inv *clibase.Invocation) error {
 			ctx := inv.Context()
 			rawURL := ""
 			if len(inv.Args) == 0 {
 				rawURL = r.clientURL.String()
+				if rawURL != "" && rawURL == inv.Environ.Get(envURL) {
+					_, _ = fmt.Fprintf(inv.Stdout, "Attempting to authenticate with environment URL: %s\n", rawURL)
```

Comment:
I wonder if we should verify the URL early, we're kind of giving the indication of success but then failing verbosely on bad input:

```
❯ CODER_URL=foo ./coder login
Attempting to authenticate with environment URL: foo
Encountered an error running "coder login"
Usage: coder login [flags] [<url>]
Error: Failed to check server "https://foo" for first user, is the URL correct and is coder accessible from your browser? Error - has initial user: Get "https://foo/api/v2/users/first": dial tcp: lookup foo on x.x.x.x:53: no such host
```

Side note: We also print `foo` only, not `https://foo` which is the default if no protocol is given.

Perhaps it might be ideal to delay the printing of messages so that we 1) note the source (`source = "config|environment|arg"`), then once we have successfully parsed the URL, we print the message noting the source.

Reply:
I like that, I like that a lot. 

Additionally, if all the options are stored as source: `"config|environment|argument|flag"` if will also give the user a useful indication as to precedence. Ex: user calls `CODER_URL=foo coder login bar` it returns `Authing via arg: bar` to indicate arg > env var.